app/controllers/locations_controller.rb

```diff
@@ -2,7 +2,11 @@ class LocationsController < ApplicationController
   def index
     expires_in Rails.application.config.cache_max_age, public: true
 
-    @locations = Location.includes(:address, :booking_location).where(state: 'current', hidden: false)
+    @locations = Location.current.includes(:address, :booking_location)
+
+    unless params[:include_hidden_locations]
+      @locations = @locations.where(hidden: false)
+    end
```

Comment:
For some reason this feels a bit difficult to parse, but I guess it works!

What about the `state: 'current'` thing, is that needed?